File: TabBarVC2/TabBarVC2/ViewController2.m
//
//  ViewController2.m
//  TabBarVC2
//
//  Created by Liu Tao on 2020/2/19.
//  Copyright © 2020 Liu Tao. All rights reserved.
//

#import "ViewController2.h"
#import "son.h"

@interface ViewController2 ()

@property (strong, nonatomic) UIButton * btn;

@end

@implementation ViewController2

+ (id)sharedInstance {
    static ViewController2 *sharedInstance = nil;
    static dispatch_once_t once;
    dispatch_once(&once, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = UIColor.brownColor;
    
    self.btn = [[UIButton alloc] initWithFrame:CGRectMake(60, 100, 50, 25)];
    self.btn.backgroundColor = UIColor.blackColor;
    [self.view addSubview:self.btn];
    [self.btn addTarget:self action:@selector(btnTapped) forControlEvents: UIControlEventTouchUpInside];
}


- (void)btnTapped {
    UIViewController *vc = [[son alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}
@end

File: TabBarVC/TabBarVC/22/ViewController2.swift
//
//  ViewController2.swift
//  TabBarVC
//
//  Created by Liu Tao on 2020/2/19.
//  Copyright © 2020 Liu Tao. All rights reserved.
//

import UIKit

class ViewController2: UIViewController {
    
    static let shared = ViewController2()
    
    let btn = UIButton.init(frame: CGRect.init(x: 50, y: 100, width: 60, height: 30))

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = UIColor.yellow
        view.addSubview(btn)
        btn.backgroundColor = UIColor.red
        btn.addTarget(self, action: #selector(btnTapped), for: UIControl.Event.touchUpInside)
    }

    @objc func btnTapped() {
        let vc = subVc2()
        navigationController?.pushViewController(vc, animated: true)
    }
}
